Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -470,6 +470,11 @@ void ComponentInfo::setDetectorInfo(DetectorInfo *detectorInfo) {
                                 "input of same size as size of DetectorInfo");
   }
   m_detectorInfo = detectorInfo;
+  /* We need to check here whether m_detectorInfo actually exists, since in the
+   * case of cloneWithoutDetectorInfo(), the detectorInfo is a null pointer.
+   */
+  if (m_detectorInfo)
+    m_detectorInfo->setComponentInfo(this);
```

Comment:
I think we probably need to better set-up links/ownership/lifetimes around these two. There are lots of ways to shoot yourself in the foot and end up with dangling pointers in one or the other of `ComponentInfo`, `DetectorInfo`. That would be future work.